test/fika/parser_test.exs

```diff
@@ -525,6 +525,19 @@ defmodule Fika.ParserTest do
     end
   end
 
+  describe "map" do
+    test "parses map with key values" do
+      str = """
+      {"a" => 1}
+      """
+
+      result = TestParser.expression!(str)
+
+      assert result ==
+               {:map, {1, 0, 10}, [{{:string, {1, 0, 4}, "a"}, {:integer, {1, 0, 9}, 1}}]}
+    end
+  end
```

Comment:
Let's add some more tests as examples of supporting other exps as keys?

Reply:
Have added some more. 

While this:
```elixir
str1 = "{&foo.bar => jar(true)}"
```
works, _considering the ones below are valid expressions as of now (since we have calls with function references)_, fails:
```elixir
str2 = "{&foo.(x) => bar(true)}"
str3 = "{&foo.bar(1) => jar(true)}"
```
with a match error
```
** (MatchError) no match of right hand side value: {:error, "expected if-else expression while processing expression in parentheses or function call or identifier or if-else expression", "{&foo.(x) => bar(true)}\n", %{}, {1, 0}, 0}
    code: result = TestParser.expression!(str)
```

Am guessing the parsing of `()` are causing an issues here. I have not added the failing ones to test.

Reply:
`&foo.(x)` is not a valid expression as per our grammar. This is because we expect users to directly call `foo(x)` instead of `&foo.(x)` in such cases. To call a function ref, users do this: `f = &foo; f.(x)`, ie, instead of directly calling on the function ref, use an identifier, return it from a function call or if-else expression etc. Does that make sense?

Reply:
Ah! 👌 
I was looking at some complex expressions that I could cook up and maybe I had it confused with this: https://github.com/fika-lang/fika/blob/426700d3537b67cfa88be29df21621b6bd6f2b4d/test/fika/parser_test.exs#L673-L691

thinking it would take args, but the test is only looking at the types rather than expressions in the args section 🤔 

>we expect users to directly call foo(x) instead of &foo.(x) in such cases.

so the same would apply to `&foo.bar(1)` case as well i suppose?